### PR TITLE
Notifications Fragmentization

### LINKED ISSUE #2248: Notifications: Convert to operate as a Fragment
To work better with future initiatives, `NotificationsActivity` should be converted to operate as a Fragment.

Also, the detail view should open in a new Activity.

### PR DESCRIPTION
**Pull request by roundhill**
Looks scary, but this PR mostly moves code out of `NotificationsActivity` and into `NotificationsListFragment` or the new `NotificationsDetailActivity`.

`NotificationsDetailListFragment` and `CommentDetailFragment` were not properly saving and restoring their state for note objects, so I fixed that as well in this PR.

Fixes #2248

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java [hunk 1]
```diff
@@ -80,366 +26,45 @@ public void onCreate(Bundle savedInstanceState) {
...
-    }
-
-    // Retrieves comment reply text so we can restore it upon returning to CommentDetailFragment
-    private String getCommentReplyText() {
-        if (mDetailFragment != null && mDetailFragment instanceof CommentDetailFragment) {
-            CommentDetailFragment commentDetailFragment = (CommentDetailFragment)mDetailFragment;
-            if (!TextUtils.isEmpty(commentDetailFragment.getReplyText())) {
-                return commentDetailFragment.getReplyText();
-            }
+        Intent intent = getIntent();
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: I love how bare-bones this activity is, but it would be even simpler if these extras could be passed when creating the fragment so no extra logic is necessary (that's going to be important in the ViewPager activity). I usually create a [newInstance](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2248-notifications-fragment/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderBlogFragment.java#L34) function in the fragment to do this, ex:

```
public static NotificationsListFragment newInstance(String noteId, boolean isReply) {
    Bundle args = new Bundle();
    args.putString(NOTE_ID_EXTRA, noteId);
    args.putBoolean(NOTE_INSTANT_REPLY_EXTRA, isReply);
    NotificationsListFragment fragment = new NotificationsListFragment();
    fragment.setArguments(args);
    return fragment;
}
```

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java [hunk 2]
```diff
@@ -80,366 +26,45 @@ public void onCreate(Bundle savedInstanceState) {
...
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
 
-        // Save text in comment reply EditText
-        if (!TextUtils.isEmpty(getCommentReplyText())) {
-            outState.putString(KEY_REPLY_TEXT, getCommentReplyText());
-        }
-
-        // Save list view scroll position
+        // This can be removed if min_sdk is 16 or higher
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: I'm not familiar with this problem. Why is this necessary?

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                    Fragment detailFragment = getDetailFragmentForNote(note);
+                    getFragmentManager().beginTransaction()
+                            .add(R.id.notifications_detail_container, detailFragment)
+                            .commit();
+
+                    if (getSupportActionBar() != null) {
+                        getSupportActionBar().setTitle(note.getTitle());
+                    }
+
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.NOTIFICATIONS_ACCESSED);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java]: I noticed we're also tracking NOTIFICATIONS_ACCESSED [here](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2248-notifications-fragment/WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java#L37). Shouldn't only one of these exist?

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java (hunk 1 above)
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: Good idea, I'll make this change.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java (hunk 2 above)
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: I'm using `startActivityForResult` from a fragment and I want to get the `onActivityResult` there as well. 

`startActivityForResult` wasn't added to `Fragment` until api 16 so I added this workaround.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java (hunk 3 above)
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java]: You're right, copy/paste fail on my part.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java (hunk 1 above)
Comment (roundhill) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: Hmm, after working on this I've remembered that we currently use `onNewIntent` to open the notifications activity with the new note if a user opens a push while notifications are already visible. So we kinda need to process the arguments in the activity still and tell the list fragment to open the note.

Open to any ideas :)

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java]: Once we switch to tabs we'll need a different solution since `onNewIntent()` won't make sense. In that situation, we'll probably want to just show the note detail when the user opens a push with no list involved (since that'll be a tab).

So...perhaps we hold off adding `newInstance()` until the redesign and for now stick with the way it currently works.

**Comment by roundhill** [comment]
@nbradbury ready for another look.

**Comment by nbradbury** [comment]
Looks good! :shipit:
